test/cql-pytest/cassandra_tests/functions/operation_fcts_test.py

```diff
@@ -0,0 +1,819 @@
+# This file was translated from the original Java test from the Apache
+# Cassandra source repository, as of commit 2a4cd36475de3eb47207cd88d2d472b876c6816d
+#
+# The original Apache Cassandra license:
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from cassandra_tests.porting import *
+
+from decimal import Decimal
+from ctypes import c_float
+from cassandra.protocol import FunctionFailure
+from cassandra.util import Date
+from datetime import datetime
+
+@pytest.mark.xfail(reason="#2693")
+def testStringConcatenation(cql, test_keyspace):
+    with create_table(cql, test_keyspace, "(a text, b ascii, c text, PRIMARY KEY(a, b, c))") as table:
+        execute(cql, table, "INSERT INTO %s (a, b, c) VALUES ('जॉन', 'Doe', 'जॉन Doe')")
+
+        with original_column_names(cql) as cql:
+            assertColumnNames(execute(cql, table, "SELECT a + a, a + b, b + a, b + b FROM %s WHERE a = 'जॉन' AND b = 'Doe' AND c = 'जॉन Doe'"),
```

Comment:
Perhaps I don't get the naming convention, but wanted to point out that first name is snake_case and the other is lowCamelCase, is that OK?

Reply:
assertColumnNames was a utility function that came from Cassandra's unit tests and I kept its lowCamelCase exactly as it was. "original_column_names" was a function that I had to add myself, to work around a Python Driver bug (or dubious feature - see my comments on original_column_names), so I chose its name in our usual coding conventions.

If you look at the oldest tests I translated in the beginning of this translation effort, I did rename all these assertColumnNames to assert_column_names() but the result was huge effort to translate each file. Things become much smoother when I just started accepting Cassandra's ugly utility function names and leaving them as is. You can see in test/cql-pytest/cassandra_test/porting.py the result of this changed decision: I have for each utility function two aliases - one in camelCase and one in snake_case.